core/src/saros/session/ISarosSession.java

```diff
@@ -151,6 +151,16 @@ public void changePermission(User user, Permission permission)
    */
   public void removeListener(ISessionListener listener);
 
+  /**
+   * Notifies the session that a shared project is now disposed or soon will be disposed. This can
+   * be used to tell the session to drop activities for no longer open projects.
+   *
+   * @param project the (disposed) shared project
+   */
+  default void disposeProject(IProject project) {
```

Comment:
What happens if you just remove the project from the sharing ? I guess there will be inconsistency warnings and I think there is also no broadcast.

Reply:
On the other hand as you are going to shutdown the session anyway, why do you not simply enable queuing ?

Reply:
Ok nevermind, at least queuing will not work as it does not affect the queue in the ActivityHandler.

I would still prefer to change the name of this method. In general you do not want to dispose projects but initiate a half ordered shutdown (discarding pending activities). 

Do you have a good name proposal ?

Reply:
How about something along the lines of `notifyProjectDisposal` or `addActivityFilterForProject`?

As you have already mentioned, this does not have to mean that the session ends. Closing projects could could also be handled by removing the closed projects from the session.
But this should be handled by other parts of the implementation. This is only a workaround to stop the session from trying to apply activities to resources that are no longer available.

Reply:
Changed the wording to reflect the filter nature.